whois-nrtm4/src/main/java/net/ripe/db/nrtm4/generator/UpdateNotificationFileGenerator.java

```diff
@@ -78,7 +85,10 @@ public void generateFile() {
 
           final List<DeltaFileVersionInfo> deltaFiles = deltaFileDao.getAllDeltasForSourceSince(nrtmSource, oneDayAgo);
           final NrtmVersionInfo fileVersion = getVersion(deltaFiles, snapshotFile.get());
-          final String json = getPayload(snapshotFile.get(), deltaFiles, fileVersion, createdTimestamp);
+
+          final NrtmKeyRecord nextKey =  nrtmKeyPairService.generateOrRotateNextKey();
```

Comment:
Can you move key rotation into a separate daily scheduled task? It's not related to the Update Notification File generator.

Reply:
Also add WARN logging so the team is aware that a key rotation is happening